src/main/java/com/launchdarkly/client/StreamProcessor.java

```diff
@@ -20,12 +25,15 @@
   private static final String INDIRECT_PUT = "indirect/put";
   private static final String INDIRECT_PATCH = "indirect/patch";
   private static final Logger logger = LoggerFactory.getLogger(StreamProcessor.class);
+  private static final int DEAD_CONNECTION_INTERVAL_SECONDS = 300;
 
   private final FeatureStore store;
   private final LDConfig config;
   private final String sdkKey;
   private final FeatureRequestor requestor;
-  private EventSource es;
+  private final ScheduledExecutorService heartbeatDetectorService;
+  private DateTime lastHeartbeat;
```

Comment:
i think we may want lastHeartbeat to be volatile, or an atomic reference. thoughts?